utils/forest_utils/src/io/writer_checksum.rs

```diff
@@ -106,11 +106,40 @@ impl<D: Digest> Checksum<D> for VoidAsyncWriterWithNoChecksum<D> {
 
 #[cfg(test)]
 mod test {
+    use anyhow::ensure;
+    use rand::{rngs::OsRng, RngCore};
     use sha2::{Sha256, Sha512};
     use tokio::io::{AsyncWriteExt, BufWriter};
 
     use super::*;
 
+    #[tokio::test]
+    async fn file_writer_tokio_fs_buf_writer() -> anyhow::Result<()> {
+        let temp_file_path = tempfile::Builder::new().tempfile()?;
+        let temp_file = tokio::fs::File::create(temp_file_path.path()).await?;
+        let mut temp_file_writer =
+            AsyncWriterWithChecksum::<Sha256, _>::new(tokio::io::BufWriter::new(temp_file));
```

Comment:
@LesnyRumcajs So this is a minimal repro of the checksum missmatch problem. The UT passes w/o `BufWriter` but fails with it.